src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/PhillipsBrineDensity.cpp

```diff
@@ -93,7 +131,21 @@ TableFunction const * makeDensityTable( string_array const & inputParams,
   }
 
   array1d< real64 > densities( tableCoords.nPressures() * tableCoords.nTemperatures() );
-  calculateBrineDensity( tableCoords, salinity, densities );
+  if( !isZero( salinity ) )
+  {
+    // if we are in the range of validity of the Phillips method, everything is good
+    // if we are not, we issue a warning message
+    calculateBrineDensity( tableCoords, salinity, densities );
+    GEOS_LOG_RANK_0_IF( salinity < 0.25,
+                        GEOS_FMT( "{}: Warning! The salinity value of {} is below the range of validity of the Phillips model, results may be inaccurate",
+                                  functionName, salinity ) );
+  }
+  else
+  {
+    // the Phillips correlation is inaccurate in the absence of salinity.
+    // since this is a very frequent case, we implement an alternate (more accurate) method below
+    calculatePureWaterDensity( tableCoords, functionName, densities );
+  }
```

Comment:
I think I would still issue a message here that we are switching to pure water properties. There might be quite a jump in the density between zero salinity and a very small salinity value which some users may find unexplained.